src/main/java/org/walkmod/javalang/compiler/types/TypesLoaderVisitor.java

```diff
@@ -135,22 +135,27 @@ public TypesLoaderVisitor(SymbolTable symbolTable, SymbolActionProvider actionPr
     * @param imported
     *           if it appears as an import declaration. It is important, because the simple name is
     *           the part after the $ symbol if it is an inner class.
+    * @param importedName
+    *           the name of the import to resolve nested classes against (optional)
     * @return the simple name to be pushed into the symbol table
     */
-   private String getSymbolName(String name, boolean imported) {
+   //@VisibleForTesting
+   static String registrationSymbolName(final String name, final boolean imported, String importedName) {
```

Comment:
I see. You have refactorized it as an static method because it do not depend on any object. I am not in love with the new name(`registrationSymbolName`). What do you think about `toSymbolName` ?